### PR TITLE
Tidy up tests directory

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Replacement for #3420

### CODE REVIEW

File: tests/auxil/asyncio.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
```
Comment (harshil21) [inline tests/auxil/asyncio.py:1]: is it a good idea to name this file as `asyncio`? Maybe `asyncio_extension`/`async_helpers`

File: tests/auxil/ci_bots.py [hunk 2]
```diff
@@ -1,4 +1,23 @@
 #!/usr/bin/env python
+
+#
+#  A library that provides a Python interface to the Telegram Bot API
```
Comment (harshil21) [inline tests/auxil/ci_bots.py:4]: this copyright has been duplicated

File: tests/auxil/networking.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+from httpx import AsyncClient, Response
+
+from telegram._utils.defaultvalue import DEFAULT_NONE
+from telegram._utils.types import ODVInput
+from telegram.error import BadRequest, RetryAfter, TimedOut
+from telegram.request import HTTPXRequest, RequestData
+
+
+class NonchalantHttpxRequest(HTTPXRequest):
+    """This Request class is used in the tests to supress errors that we don't to care about
```
Comment (harshil21) [inline tests/auxil/networking.py:31]:
```suggestion
    """This Request class is used in the tests to suppress errors that we don't want to care about
```

**Review by harshil21 — COMMENTED** [review]
Nice work cleaning up the tests structure. Maybe worth doing these as well:

- Adding the `ext`, `inline`, `passport` folders in the tests so we can shift related test files in there. I believe they can have their own `conftest`s as well if that helps in better organizing it

File: tests/auxil/ci_bots.py (hunk 2 above)
Comment (Bibo-Joshi) [reply tests/auxil/ci_bots.py:4]: damn, thought I got all …

**Comment by Bibo-Joshi** [comment]
> * Adding the `ext`, `inline`, `passport` folders in the tests so we can shift related test files in there. I believe they can have their own `conftest`s as well if that helps in better organizing it

I guess then it would make sense to mimic the whole directory structure of `telegram`. I'll have to look more closely into how pytest discovers tests. should have time on the weekend

**Review by harshil21 — APPROVED** [review]
LGTM

**Pull request merged**